src/kinetics/ReactionRate.cpp

```diff
@@ -0,0 +1,44 @@
+//! @file RxnRates.cpp
+
+// This file is part of Cantera. See License.txt in the top-level directory or
+// at https://cantera.org/license.txt for license and copyright information.
+
+#include "cantera/kinetics/ReactionRate.h"
+#include "cantera/numerics/Func1.h"
+#include "cantera/thermo/ThermoPhase.h"
+
+namespace Cantera
+{
+
+void CustomFunc1Data::update(const ThermoPhase& bulk) {
+    m_temperature = bulk.temperature();
+}
+
+CustomFunc1Rate::CustomFunc1Rate() : m_ratefunc(0) {}
+
+void CustomFunc1Rate::setRateFunction(shared_ptr<Func1> f) {
+    m_ratefunc = f;
+}
+
+double CustomFunc1Rate::eval(const CustomFunc1Data& shared_data) const {
+    if (m_ratefunc) {
+        return m_ratefunc->eval(shared_data.m_temperature);
+    }
+    return 0.;
```

Comment:
Should this throw if no rate function is set? Just returning 0 if you forget to set the rate function seems a little too error-prone.

Reply:
I chose that behavior to be consistent with other default constructor behavior (e.g. the original `Arrhenius` will create zero rates if no arguments are passed). I.e. there could be an argument made to raise an error in all cases.

Reply:
Fair enough.